lib/livebook/hubs/team_client.ex

```diff
@@ -483,12 +532,14 @@ defmodule Livebook.Hubs.TeamClient do
     deployment_group_id = app_deployment.deployment_group_id
 
     with {:ok, deployment_group} <- fetch_deployment_group(deployment_group_id, state) do
+      state = put_app_deployment(state, app_deployment)
+      Teams.Broadcasts.app_deployment_created(app_deployment)
+
       if deployment_group.id == state.deployment_group_id do
-        :ok = download_and_deploy(state.hub, app_deployment, state.derived_key)
+        manager_sync()
       end
```

Comment:
So currently on `:agent_connected` we call `dispatch_app_deployments` and this call the event handler for every app deployment. Ideally we would decouple it such that we call `manager_sync` once on `:agent_connected` and once on future app deployment events.